### PR TITLE
Egd 7629 reedgefs

### PR DESCRIPTION
**Pull request by alekrudnik**
Added support for reliance edge file system.
    The initial port is done.
    The whole effort is still WIP.
    Left a note in reamde file what should be done next.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -99,3 +99,7 @@
 	path = third-party/lwext4/lwext4
 	url = ../lwext4.git
 	branch = mudita
+[submodule "third-party/reedgefs/src"]
```
[the remark this answers is not in the document]
Comment (alekrudnik) [reply .gitmodules:102]: done

File: third-party/reedgefs/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,36 @@
+
+add_library(reedgefs INTERFACE)
```
[the remark this answers is not in the document]
Comment (alekrudnik) [reply third-party/reedgefs/CMakeLists.txt:2]: it is how the porting of reedgefs must be done. Some sources are definitions are in our code, and some are here.

File: third-party/reedgefs/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+add_library(reliance-edge::fs ALIAS reedgefs)
+
+target_include_directories(reedgefs
+    INTERFACE
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/include>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/core/include>
+)
+
+target_sources( reedgefs
```
[the remark this answers is not in the document]
Comment (alekrudnik) [reply third-party/reedgefs/CMakeLists.txt:11]: done

File: module-platform/linux/tests/CMakeLists.txt [hunk 4]
```diff
@@ -94,6 +94,16 @@ add_catch2_executable(
     USE_FS
 )
 
+add_catch2_executable(
+    NAME vfs-reedgefs
+    SRCS
+        ${CMAKE_CURRENT_LIST_DIR}/unittest_filesystem_reedgefs.cpp
```
[the remark this answers is not in the document]
Comment (alekrudnik) [reply module-platform/linux/tests/CMakeLists.txt]: fixed

**Review by niepiekm — APPROVED** [review]

**Review by lucckb — APPROVED** [review]

File: .gitmodules [hunk 5]
```diff
@@ -102,3 +102,7 @@
 [submodule "third-party/dr_libs/src"]
 	path = third-party/dr_libs/src
 	url = https://github.com/mudita/dr_libs.git
+[submodule "reliance-edge"]
+	path = third-party/reedgefs/src
+	url = ../reliance-edge.git
```
Comment (SP2FET) [inline .gitmodules:107]: could you please use full URL for submodule?

**Review by SP2FET — APPROVED** [review]

File: .gitmodules (hunk 5 above)
Comment (alekrudnik) [reply .gitmodules:107]: Most of mudita forkes got relative paths.

If we need to use full paths let's cover all in separate jira case

**Pull request merged**